codegen/Generate.elm

```diff
@@ -38,29 +38,43 @@ moduleDocsToFile moduleDocs =
         moduleName : List String
         moduleName =
             moduleDocs.name |> String.split "."
+
+        values : List String
+        values =
+            List.map .name moduleDocs.values
     in
     Elm.file ("Fn" :: moduleName)
-        ((moduleDocs.values |> List.map .name)
+        (List.map (\name -> declaration moduleName ( name, name )) values
             ++ (moduleDocs.unions
-                    |> List.concatMap
-                        (\union ->
-                            union.tags
-                                |> List.map (\( variantName, _ ) -> variantName)
+                    |> List.concatMap .tags
+                    |> List.map
+                        (\( tagName, _ ) ->
+                            declaration moduleName ( tagDeclarationName values (stringFirstCharToLower tagName), tagName )
                         )
                )
-            |> List.map
-                (\name ->
-                    Elm.declaration (name |> stringFirstCharToLower)
-                        (Elm.tuple
-                            (Elm.list (moduleName |> List.map Elm.string)
-                                |> Elm.withType Gen.Elm.Syntax.ModuleName.annotation_.moduleName
-                            )
-                            (Elm.string name)
-                        )
-                )
         )
 
 
+declaration : List String -> ( String, String ) -> Elm.Declaration
+declaration moduleName ( declarationName, originalName ) =
+    Elm.declaration declarationName
+        (Elm.tuple
+            (Elm.list (moduleName |> List.map Elm.string)
+                |> Elm.withType Gen.Elm.Syntax.ModuleName.annotation_.moduleName
+            )
+            (Elm.string originalName)
+        )
+
+
+tagDeclarationName : List String -> String -> String
```

Comment:
I'd be fine with suffixing _every_ variant or even creating e.g. `Fn.Result.Variant.err`

Reply:
Yeah, that's probably a good idea, less confusing :+1:

Reply:
We'll end up with things like `Fn.Maybe.justVariant` and `Fn.Basics.trueVariant`. Does that sound okay to you? I'm fine with it, but it is definitely not as nice for the common case. In fact, I don't know if we'll ever run into a case where we'll need to refer to a variant that would otherwise result in a conflict (at least today we don't).

Reply:
sounds good I think.

Reply:
Alright, let's try that then. We can always undo this later otherwise.